### PR TITLE
Add Firework meta

### PR DESCRIPTION
**Pull request by ZephireNZ**
Just in time for Guy Fawkes!

Includes the necessary handling of ItemMeta for Firework charges and Fireworks.

Yep, that's about it.

---

_Edited by turt2live_

**Related Links:**
- Ticket: #77

### CODE REVIEW

**Comment by dequis** [comment]
Ohhh, and recipes for this in the other PR. Awesome.

(also, checkstyle)

**Comment by ZephireNZ** [comment]
Motherfu-

**Comment by dequis** [comment]
Referencing #77

**Comment by dequis** [comment]
Simple conflict in GlowItemFactory, keep both sides. [Branch](https://github.com/dequis/Glowstone/compare/pr-506). This one is your fault too (skulls and banners), and this means it will conflict with #496.

**Comment by turt2live** [comment]
@ZephireNZ When you get the chance, please merge @dequis's changes (or do it yourself). Thanks! 

Edit: Now known that @ZephireNZ's PC is a paperweight as per https://github.com/GlowstoneMC/Glowstone/pull/496#issuecomment-62086624

**Comment by ZephireNZ** [comment]
Can't at the moment, computer is broke. Could be quite a while before I get
the chance, sadly.

**Comment by dequis** [comment]
Threw two PRs at @ZephireNZ because why not

File: src/main/java/net/glowstone/inventory/GlowMetaCharge.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+import org.bukkit.Color;
+import org.bukkit.FireworkEffect;
+import org.bukkit.Material;
+import org.bukkit.inventory.meta.FireworkEffectMeta;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class GlowMetaCharge extends GlowMetaItem implements FireworkEffectMeta {
```
Comment (turt2live) [inline src/main/java/net/glowstone/inventory/GlowMetaCharge.java]: I would suggest calling this GlowMetaFireworkEffect to remain consistent

Comment (dequis) [inline src/main/java/net/glowstone/inventory/GlowMetaCharge.java]: `GlowMetaFireworkCharge`?

Comment (turt2live) [inline src/main/java/net/glowstone/inventory/GlowMetaCharge.java]: It's specific to charges right now, but may not be in the future.

Comment (ZephireNZ) [inline src/main/java/net/glowstone/inventory/GlowMetaCharge.java]: Considering the Bukkit meta it backs is FireworkEffect, I will probably
stick with that. Thanks for spotting :)

Comment (turt2live) [inline src/main/java/net/glowstone/inventory/GlowMetaCharge.java]: I've submitted a pull request to change this considering your PC is dead <3

https://github.com/ZephireNZ/Glowstone/pull/5

Once accepted (or a similar change done), this is ready to be pulled :3

**Comment by SpaceManiac** [comment]
Merged in 3a7e400f1ce49ca72c7311d28dc6f99ef0991924.

**Pull request closed without merging**